lib/scripting/post-process/class.ts

```diff
@@ -0,0 +1,218 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { replace } from 'estraverse';
+import {
+	BlockStatement,
+	ClassBody,
+	FunctionDeclaration,
+	Identifier,
+	MethodDefinition,
+	Statement,
+	VariableDeclaration,
+} from 'estree';
+import {
+	assignmentExpression,
+	blockStatement,
+	classBody,
+	classDeclaration,
+	expressionStatement,
+	functionExpression,
+	identifier,
+	memberExpression,
+	methodDefinition,
+	returnStatement,
+	thisExpression,
+	variableDeclaration,
+	variableDeclarator,
+} from '../estree';
+import { ESIToken } from '../grammar/grammar';
+
+export function ppClass(node: ESIToken): any {
+	let estree = null;
+	if (node.type === 'ClassDeclaration') {
+		estree = ppClassDeclaration(node);
+	} else if (node.type === 'ConstructorMemberDeclaration') {
+		estree = ppConstructorMemberDeclaration(node);
+	} else if (node.type === 'RegularPropertyMemberDeclaration') {
+		estree = ppRegularPropertyMemberDeclaration(node);
+	} else if (node.type === 'PropertyGetDeclaration') {
+		estree = ppPropertyGetDeclaration(node);
+	} else if (node.type === 'PropertyLetDeclaration') {
+		estree = ppPropertyLetDeclaration(node);
+	} else if (node.type === 'PropertySetDeclaration') {
+		estree = ppPropertySetDeclaration(node);
+	}
+	return estree;
+}
+
+function ppClassDeclaration(node: ESIToken): any {
+	let id = identifier('undefined');
+	let constructor: MethodDefinition | undefined;
+	const methodDefinitions: MethodDefinition[] = [];
+	const varStmts: Statement[] = [];
+	const ids: string[] = [];
+	for (const child of node.children) {
+		if (child.type === 'Identifier') {
+			id = child.estree;
+		} else if (child.type === 'ClassMemberDeclaration') {
+			const memberDecl = child.children[0];
+			if (memberDecl.type === 'ConstructorMemberDeclaration') {
+				constructor = memberDecl.estree;
+			} else if (memberDecl.type === 'MethodMemberDeclaration') {
+				const functionDecl = memberDecl.estree as FunctionDeclaration;
+				const functionId = functionDecl.id as Identifier;
+				methodDefinitions.push(
+					methodDefinition(functionId, 'method', functionExpression(functionDecl.body, functionDecl.params)),
+				);
+				ids.push(functionId.name);
+			} else if (memberDecl.type === 'PropertyMemberDeclaration') {
+				methodDefinitions.push(memberDecl.estree);
+			} else if (
+				memberDecl.type === 'VariableMemberDeclaration' ||
+				memberDecl.type === 'ConstantMemberDeclaration'
+			) {
+				for (const varDecl of (memberDecl.estree as VariableDeclaration).declarations) {
+					const varId = varDecl.id as Identifier;
+					varStmts.push(
+						expressionStatement(
+							assignmentExpression(
+								memberExpression(thisExpression(), varId),
+								'=',
+								varDecl.init ? varDecl.init : identifier('undefined'),
+							),
+						),
+					);
+					ids.push(varId.name);
+				}
+			}
```

Comment:
This applies to many other blocks too: Firstly, a `switch` statement would be more readable, and printing a warning when none of the cases is met would probably help you debugging more quickly in the future?